commands/scp.go

```diff
@@ -148,6 +150,11 @@ func getInfoForScpArg(hostAndPath string, hostInfoLoader HostInfoLoader) (HostIn
 	}
 
 	args := []string{}
+	port, err := hostInfo.GetSSHPort()
+	if err == nil && port > 0 {
+		args = append(args, "-P", fmt.Sprintf("%v", port))
+	}
```

Comment:
Indeed it'd work when it's between local and a remote, I was just pointing out an edge case which wouldn't be supported. Your PR is still an improvement :+1: